packages/asset-swapper/contracts/src/BalancerV2Sampler.sol

```diff
@@ -109,6 +109,47 @@ contract BalancerV2Sampler is SamplerUtils {
         }
     }
 
+    // TODO - THIS IS PSEUDO CODE AS A DEMO
+    // Replaces amount for first step with each takerTokenAmount and calls queryBatchSwap using supplied steps
+    /// @dev Sample sell quotes from Balancer V2 supporting multihops.
+    /// @param swapSteps Array of swap steps (can be >= 1).
+    /// @param swapAssets Array of token address for swaps.
+    /// @param takerTokenAmounts Taker token sell amount for each sample.
+    function sampleMultihopSellsFromBalancerV2(
+        IBalancerV2Vault.BatchSwapStep[] memory swapSteps,
+        IAsset[] memory swapAssets,
+        uint256[] memory takerTokenAmounts
+    )
+        public
+        returns (uint256[] memory makerTokenAmounts)
+    {
+        IBalancerV2Vault vault = IBalancerV2Vault(poolInfo.vault);
+        uint256 numSamples = takerTokenAmounts.length;
+        makerTokenAmounts = new uint256[](numSamples);
+        IBalancerV2Vault.FundManagement memory swapFunds =
+            _createSwapFunds();
+
+        for (uint256 i = 0; i < numSamples; i++) {
+            swapSteps[0].amount = takerTokenAmounts[i];
+            try
+                // For sells we specify the takerToken which is what the vault will receive from the trade
+                vault.queryBatchSwap(IBalancerV2Vault.SwapKind.GIVEN_IN, swapSteps, swapAssets, swapFunds)
+            // amounts represent pool balance deltas from the swap (incoming balance, outgoing balance)
+            returns (int256[] memory amounts) {
+                // Outgoing balance is negative so we need to flip the sign
+                // TO DO - Add index for asset out
```

Comment:
What do you mean by this?

Reply:
I may be reading it wrong but previously queryBatchSwap was only returning two deltas - for tokenIn and tokenOut, because it was always a single swap and there were only two tokens involved.
For longer paths queryBatchSwap will return a delta for each token in the assets array and we want the delta for the last index which should be tokenOut.